### ISSUE TITLE
Parameters for experiments in `Bag of Tricks`

### ISSUE THREAD

**Issue by bkj** [facebookresearch/fastText#136]
Hi all -- are you able to release the set of parameters that you used for Table 3 in the 'Bag of Tricks' paper?  I'm trying to reproduce the results and it's not totally clear from the paper how you did the parameter grid search?

Thanks!

**Comment by ajoulin** [comment]
Hi,

Unfortunately, I didn't save the logs of these experiments. 

From memory, I think that we used a grid search of parameters (-epoch, -lr, -dim, -bucket, max_ngram_size). We used the validation set to pick the best set of hyper-parameters for each dataset (see the data at http://ir.hit.edu.cn/~dytang/). I think the range for the hyperparameters were: epochs in {1, 2, 5, 10}, lr in {0.05,  0.1,  0.25,  0.5} ngram in {1,2,3,4,5}, dim in {10,50,100},  bucket in {1M, 2M, 10M}. 

Let me know if you have an issue reproducing some of the numbers. 

Sorry for the inconvenience.

**Comment by bkj** [comment]
Ah no problem -- these kinds of things often get lost in the shuffle.  I
did a somewhat similar grid search and was able to get somewhat similar
numbers, but I'll try to reproduce the search more precisely.

Also, do you recall if you did any preprocessing of the data beyond what's
aviabable on Tang's site?

**Comment by bkj** [comment]
Looks like I was able to get pretty close w/ that grid search on Yelp '13 -- these are the top 10 dev set accuracies w/ corresponding test errors:

```
     1	{"dev":0.6403689442106206,"test":0.6403116045845272}
     2	{"dev":0.6402196949344796,"test":0.6416845749761223}
     3	{"dev":0.6400405958031103,"test":0.6416547277936963}
     4	{"dev":0.6398316468165129,"test":0.6409085482330468}
     5	{"dev":0.6395331482642309,"test":0.640639923591213}
     6	{"dev":0.6395032984090027,"test":0.6402817574021012}
     7	{"dev":0.6393241992776335,"test":0.6409682425978988}
     8	{"dev":0.6392644995671771,"test":0.642042741165234}
     9	{"dev":0.6389660010148951,"test":0.6398340496657116}
    10	{"dev":0.6389361511596668,"test":0.6369090257879656}
```

I used multiple threads, so that introduces some stochasticity -- ideally I guess this grid search should be done on a single thread, but that would take even longer.  

For posterity, the top 3 performing parameter settings for Yelp '13 (no preprocessing beyond what Tang did):

```
#1)  {
  "loss": "softmax",
  "thread": 2,
  "input_file": "yelp-2013-train-ft.txt",
  "dev_results": {
    "recall": 0.6403689442106206,
    "precision": 0.6403689442106206
  },
  "word_ngrams": 5,
  "bucket": 10000000,
  "test_results": {
    "recall": 0.6403116045845272,
    "precision": 0.6403116045845272
  },
  "epoch": 10,
  "lr": 0.1,
  "dim": 50,
}
#2)  {
  "loss": "softmax",
  "thread": 2,
  "input_file": "yelp-2013-train-ft.txt",
  "dev_results": {
    "recall": 0.6402196949344796,
    "precision": 0.6402196949344796
  },
  "word_ngrams": 5,
  "bucket": 10000000,
  "test_results": {
    "recall": 0.6416845749761223,
    "precision": 0.6416845749761223
  },
  "epoch": 5,
  "lr": 0.25,
  "dim": 50,
}
#3)  {
  "loss": "softmax",
  "thread": 2,
  "input_file": "yelp-2013-train-ft.txt",
  "dev_results": {
    "recall": 0.6400405958031103,
    "precision": 0.6400405958031103
  },
  "word_ngrams": 5,
  "bucket": 10000000,
  "test_results": {
    "recall": 0.6416547277936963,
    "precision": 0.6416547277936963
  },
  "epoch": 10,
  "lr": 0.1,
  "dim": 100,
}
```

**Comment by ajoulin** [comment]
I m surprised that with 2 threads you only get to 0.6403, but overall the numbers seem consistent. To be fair, I should have run these experiments with different seeds, and give the average performance with error bars.

For completeness, I think I run the exp. on 20 threads with a code that have been changed since then. This may have introduced additional stochasticity: the matrices were initialize in a different order for instance and I didn't fix the seed (I believe that now we do).

Once I ll have a bit of time, I ll rerun these exp. with the latest version of the code and check if there is some differences in the numbers and update the paper accordingly .

**Comment by cpuhrsch** [comment]
I'm closing this due to age, but please feel free to reopen it at any time if this issue is still current.
